### PR TITLE
Sentry reporting + cleanup

### PR DESCRIPTION
**Pull request by ThaTiemsz**
Sentry reporting :tada: plus some cleaning up.
I'd recommend testing it first, of course :roll_eyes:

### CODE REVIEW

**Comment by devsnek** [comment]
as long as you're doing this can you chain all those promises instead of nesting them

**Comment by ThaTiemsz** [comment]
@devsnek You have write access, do it yourself 😄

File: Utils/error_loggers.js [hunk 1]
```diff
@@ -1,9 +1,14 @@
...
-exports.log = function (bot, cObj, fullErr) {
-  if (fullErr !== undefined) bugsnag.notify(fullErr)
-  bot.Channels.find((c) => c.name === 'bot-error').sendMessage(`Encountered an error while trying to run ${cObj.cause}.\nReturned error: \`\`\`${cObj.message}\`\`\``)
-}
+module.exports = {
+  log: (bot, cObj, fullErr) => {
+    if (fullErr !== undefined) Raven.captureException(fullErr)
+    bot.Channels.find((c) => c.name === 'bot-error').sendMessage(`Encountered an error while trying to run ${cObj.cause}.\nReturned error: \`\`\`${cObj.message}\`\`\``)
+  },
+  raven: (e) => {
```
Comment (devsnek) [inline Utils/error_loggers.js]: you could add an "extra" param, which will get pushed to sentry for additional info
```js
raven: (error, extra) => Raven.captureException(error, { extra });
```

**Comment by LW001** [comment]
I did some testing, looks good from what I can tell.

File: Commands/Commands/uservoice.js [hunk 2]
```diff
@@ -1,10 +1,10 @@
-var commands = []
+let commands = []
 
-// var checker = require('../../Utils/access_checker')
-var logger = require('../../Utils/error_loggers')
-var config = require('../../config.js')
-var Entities = require('html-entities').AllHtmlEntities
-var UVRegex = /https?:\/\/[\w.]+\/forums\/(\d{6,})-[\w-]+\/suggestions\/(\d{7,})(?:-[\w-]*)?/
+// const checker = require('../../Utils/access_checker')
```
Comment (Arzte) [inline Commands/Commands/uservoice.js:3]: If your moving variables like these over to ``const``'s, doesn't that make those variable program wide? So why repeat them, and as ``const`` after moving them to that?

File: Commands/Commands/uservoice.js [hunk 3]
```diff
@@ -308,7 +308,7 @@ commands.info = {
               },
               timestamp: new Date(data.user.created_at)
             })
-          }).catch(() => {}) // Make bugsnag ignore it, error handling below
```
Comment (Arzte) [inline Commands/Commands/uservoice.js:311]: Comment is there on purpose, feel free to change it to sentry, however.

File: Commands/Commands/utils.js [hunk 4]
```diff
@@ -102,11 +99,11 @@ commands.stats = {
...
           },
           fields: field
-        }).catch(bugsnag.notify) // Send Message to DM error
-      }).then(msg.delete()).catch(bugsnag.notify) // Error opening DM channel
+        }).catch(logger.raven) // Send Message to DM error
+      }).then(msg.delete()).catch(logger.raven) // Error opening DM channel
     }).catch(e => {
       msg.reply('an unexpected error occured while getting your stats, try again later.')
-      console.error(e)
+      logger.raven(e)
```
Comment (Arzte) [inline Commands/Commands/utils.js:106]: 👍 thanks for cleaning this up

File: Commands/Commands/queue.js [hunk 5]
```diff
@@ -1,17 +1,14 @@
...
 const r = new Dash()
-var checker = require('../../Utils/access_checker')
-var logger = require('../../Utils/error_loggers')
-var genlog = require('../../Utils/generic_logger')
-var config = require('../../config.js')
-var bugsnag = require('bugsnag')
+const checker = require('../../Utils/access_checker')
+const logger = require('../../Utils/error_loggers')
+const genlog = require('../../Utils/generic_logger')
+const config = require('../../config.js')
```
Comment (SirDoctors) [inline Commands/Commands/queue.js]: see my other comment on this.

File: Commands/Commands/top10.js [hunk 6]
```diff
@@ -1,10 +1,10 @@
-var Entities = require('html-entities').AllHtmlEntities
-var config = require('../../config.js')
+const Entities = require('html-entities').AllHtmlEntities
+const config = require('../../config.js')
```
Comment (Arzte) [inline Commands/Commands/top10.js:2]: this too

File: Commands/Commands/utils.js [hunk 7]
```diff
@@ -1,16 +1,13 @@
...
 
-// var checker = require('../../Utils/access_checker')
-// var logger = require('../../Utils/error_loggers')
-var config = require('../../config.js')
-var analytics = require('../../Utils/orwell.js')
-var bugsnag = require('bugsnag')
+// const checker = require('../../Utils/access_checker')
+const logger = require('../../Utils/error_loggers')
+const config = require('../../config.js')
+const analytics = require('../../Utils/orwell.js')
```
Comment (SirDoctors) [inline Commands/Commands/utils.js]: and these

File: Utils/dbinit.js [hunk 8]
```diff
@@ -1,5 +1,5 @@
-var Dash = require('rethinkdbdash')
-var r = new Dash()
+const Dash = require('rethinkdbdash')
+const r = new Dash()
```
Comment (Arzte) [inline Utils/dbinit.js:2]: and this

File: Utils/error_loggers.js [hunk 9]
```diff
@@ -1,9 +1,14 @@
 const Config = require('../config.js')
-var bugsnag = require('bugsnag')
+const Raven = require('raven')
 
-bugsnag.register(Config.discord.bugsnag)
+Raven.config(Config.discord.sentry).install()
```
Comment (Arzte) [inline Utils/error_loggers.js]: reeee more config changes, reeee. 👍

File: Commands/Commands/uservoice.js (hunk 2 above)
Comment (ThaTiemsz) [reply Commands/Commands/uservoice.js:3]: Constants aren't program wide, they're basically variables, except you can't reassign or redeclare them.

File: Utils/error_loggers.js (hunk 9 above)
Comment (Dougley) [reply Utils/error_loggers.js]: Actually, change that to `Config.services.sentry`

File: Commands/Commands/uservoice.js (hunk 2 above)
Comment (Arzte) [reply Commands/Commands/uservoice.js:3]: alright, thanks for adding clarification on that.

File: Commands/Commands/uservoice.js [hunk 10]
```diff
@@ -308,7 +308,7 @@ commands.info = {
...
+          }).catch(logger.raven)
```
Comment (Arzte) [inline Commands/Commands/uservoice.js:311]: I was talking about the changing the wording in the previously there comment to reflect that it was ignored on purpose in that catch. Sorry if I was a bit misleading, however this should probably be reverted to ``.catch(() => {}) // Make sentry ignore it, error handling below``

File: .gitignore [hunk 11]
```diff
@@ -54,4 +54,5 @@ channels.js
 # Roles config
 roles.js
 
+package-lock.json
```
Comment (Arzte) [inline .gitignore]: If package-lock gen is turned off, this is probably redundent

File: Utils/error_loggers.js [hunk 12]
```diff
@@ -1,9 +1,12 @@
...
+const Raven = require('raven')
 
-bugsnag.register(Config.discord.bugsnag)
+Raven.config(Config.services.sentry).install()
 
-exports.log = function (bot, cObj, fullErr) {
-  if (fullErr !== undefined) bugsnag.notify(fullErr)
-  bot.Channels.find((c) => c.name === 'bot-error').sendMessage(`Encountered an error while trying to run ${cObj.cause}.\nReturned error: \`\`\`${cObj.message}\`\`\``)
-}
+module.exports = {
```
Comment (SirDoctors) [inline Utils/error_loggers.js]: Is this supposed to be changed from ``exports.log`` to module.exports?

File: Commands/Commands/uservoice.js (hunk 3 above)
Comment (Dougley) [reply Commands/Commands/uservoice.js:311]: Actually no, leave this out.

File: Commands/Commands/uservoice.js [hunk 13]
```diff
@@ -308,7 +308,7 @@ commands.info = {
               },
               timestamp: new Date(data.user.created_at)
             })
-          }).catch(() => {}) // Make bugsnag ignore it, error handling below
+          }).catch(() => {}) // Make sentry ignore it, error handling below
```
Comment (Dougley) [inline Commands/Commands/uservoice.js]: ~~Why did you listen to doctors~~

`.catch(logger.raven)`

File: index.js [hunk 14]
```diff
@@ -131,9 +110,13 @@ bot.Dispatcher.on(Events.GATEWAY_READY, () => {
...
 })
 
+process.on('uncaughtException', (err) => {
+  logger.raven(err.stack || err)
+})
+
 process.on('unhandledRejection', (reason, p) => {
   if (p !== null && reason !== null) {
-    bugsnag.notify(new Error(`Unhandled promise: ${require('util').inspect(p, {depth: 3})}: ${reason}`))
+    logger.raven(new Error(`Unhandled promise: ${require('util').inspect(p, {depth: 3})}: ${reason}`))
```
Comment (Dougley) [inline index.js]: Do a `typeof` check to make sure `p` is not an error already, if it is, send it straight to raven.

**Review by Dougley — CHANGES_REQUESTED** [review]

**Comment by devsnek** [comment]
please change the stuff so that the promises are chained

**Review by nullpixel — COMMENTED** [review]
[REMOVED]

**Comment by Arzte** [comment]
@devsnek I'll be working on promises in another pr for the entire project, if that's all then I'll goahead and merge this.

**Review by Dougley — APPROVED** [review]

**Pull request merged**